SimTKcommon/include/SimTKcommon/internal/recorder.h

```diff
@@ -47,7 +47,7 @@ class DLL_EXPORT Recorder {
   Recorder& operator = ( const Recorder& );
 
   /* IO friends */
-  friend DLL_EXPORT std::istream& operator >> (std::istream& is, const Recorder& a);
+  friend DLL_EXPORT std::istream& operator >> (std::istream& is, Recorder& a);
```

Comment:
Is this necessary? I would be careful in changing things in `Recorder`. Why cannot it stay `const`?

Reply:
If we keep const, an issue appears when compiling the function:
bool String::tryConvertToRecorder(Recorder& out) const {...} in String.cpp
This is related to the class basic_istream

In that case, we may change the structure of tryConvertToRecorder, which is similar to tryConvertToDouble, and probably many more functions.

It works well only removing "const" in Recorder.